pkg/scheduler/redis_scheduler.go

```diff
@@ -25,6 +27,10 @@ const SchedKey = "scheduling"
 // pollInterval is the time interval between 2 redis polling
 const pollInterval = 1 * time.Second
 
+// eventLoopSize is the number of goroutines handling @events and triggering
+// jobs.
+const eventLoopSize = 50
```

Comment:
I think we can benefit from some level of concurrency to handle the global event channel to perform the work done in the event loop.

This handling of each event is quite important:
  - fetching the set of triggers form redis
  - parsing all the rules
  - fetching the matching trigger from couchdb
  - sending a new job request into the queue

Even with some caching, having some fixed size concurrency to handle that seems beneficial to me. Regarding the amount of go-routine, we could make it configurable.